server/src/main/java/org/jboss/as/server/deployment/ExplodedDeploymentAddContentHandler.java

```diff
@@ -0,0 +1,181 @@
+        final byte[] oldHash = CONTENT_HASH.resolveModelAttribute(context, contentItemNode).asBytes();
+        final boolean overwrite = OVERWRITE.resolveModelAttribute(context, contentItemNode).asBoolean(true);
+        List<ModelNode> contents = EXPLODED_CONTENT.resolveModelAttribute(context, operation).asList();
+        final List<ExplodedContent> addedFiles = new ArrayList<>(contents.size());
+        final byte[] newHash;
+        if (contents.size() == 1 && contents.get(0).hasDefined(HASH)) {
+            newHash = DeploymentHandlerUtil.addFromHash(contentRepository, contents.get(0), managementName, address, context);
+            if (operation.hasDefined(DeploymentAttributes.UPDATED_PATHS.getName())) {
+                for (ModelNode addedFile : DeploymentAttributes.UPDATED_PATHS.resolveModelAttribute(context, operation).asList()) {
+                    addedFiles.add(new ExplodedContent(addedFile.asString()));
+                }
+            }
+        } else {
+            for (ModelNode content : contents) {
+                InputStream in = DeploymentHandlerUtils.getInputStream(context, content);
+                String path = TARGET_PATH.resolveModelAttribute(context, content).asString();
+                addedFiles.add(new ExplodedContent(path, in));
+            }
+            try {
+                newHash = contentRepository.addContentToExploded(oldHash, addedFiles, overwrite);
+            } catch (ExplodedContentException e) {
+                throw createFailureException(e.toString());
+            }
+        }
+        final List<String> relativePaths = addedFiles.stream().map(ExplodedContent::getRelativePath).collect(Collectors.toList());
+        contentItemNode.get(CONTENT_HASH.getName()).set(newHash);
+        contentItemNode.get(CONTENT_ARCHIVE.getName()).set(false);
+        if (!addedFiles.isEmpty() && ENABLED.resolveModelAttribute(context, deploymentResource.getModel()).asBoolean()) {
+            context.addStep(new OperationStepHandler() {
+                @Override
+                public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                    try {
+                        ExecutorService executor = (ExecutorService) context.getServiceRegistry(false).getRequiredService(JBOSS_SERVER_EXECUTOR).getValue();
+                        CountDownLatch latch = copy(executor, relativePaths, managementName, newHash);
+                        if (latch != null) {
+                            try {
+                                if (!latch.await(60, TimeUnit.SECONDS)) {
+                                    throw ServerLogger.ROOT_LOGGER.couldNotCopyFiles();
+                                }
+                            } catch (InterruptedException e) {
+                                Thread.currentThread().interrupt();
+                                throw createFailureException(e.toString());
```

Comment:
Don't throw an exception, just return. You restored the interrupted status on the thread, so the OperationContext will detect that and go into its operation cancellation logic. Throwing an exception just adds noise to that.